Gateway/Http/TransactionBuilder/RefundPartial.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+/**
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the MIT License
+ * It is available through the world-wide-web at this URL:
+ * https://tldrlegal.com/license/mit-license
+ * If you are unable to obtain it through the world-wide-web, please send an email
+ * to support@buckaroo.nl so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade this module to newer
+ * versions in the future. If you wish to customize this module for your
+ * needs please contact support@buckaroo.nl for more information.
+ *
+ * @copyright Copyright (c) Buckaroo B.V.
+ * @license   https://tldrlegal.com/license/mit-license
+ */
+
+namespace Buckaroo\Magento2\Gateway\Http\TransactionBuilder;
+
+use Magento\Store\Model\Store;
+use Magento\Framework\App\RequestInterface;
+use Buckaroo\Magento2\Model\GroupTransaction;
+use Magento\Framework\HTTP\PhpEnvironment\RemoteAddress;
+
+
+
+class RefundPartial extends AbstractTransactionBuilder
+{
+
+    /**
+     * @var \Magento\Framework\App\RequestInterface
+     */
+    protected $httpRequest;
+
+
+    /**
+     * @var \Magento\Store\Model\Store
+     */
+    protected $store;
+
+
+    /**
+     *
+     * @var \Buckaroo\Magento2\Model\GroupTransaction
+     */
+    protected $groupTransaction;
+
+    /**
+     * Set giftcard transaction
+     *
+     * @param GroupTransaction $groupTransaction
+     *
+     * @return self
+     */
+    public function setGroupTransaction(GroupTransaction $groupTransaction)
+    {
+        $this->groupTransaction = $groupTransaction;
+        return $this;
+    }
+
+    /**
+     * Set originating http request
+     *
+     * @param RequestInterface $request
+     *
+     * @return self
+     */
+    public function setRequest(RequestInterface $request)
+    {
+        $this->httpRequest = $request;
+        return $this;
+    }
+
+    /**
+     * Set current store
+     *
+     * @param Store $store
+     *
+     * @return self
+     */
+    public function setStore(Store $store)
+    {
+        $this->store = $store;
+        return $this;
+    }
+    /**
+     * @return array
+     */
+    public function getBody()
+    {
+        if (!$this->store instanceof Store) {
+            throw new \Exception("`store` must be instance of Magento\Store\Model\Store");
+        }
+
+
+        if (!$this->groupTransaction instanceof GroupTransaction) {
+            throw new \Exception("`groupTransaction` must be instance of Buckaroo\Magento2\Model\GroupTransaction");
+        }
+
+        $ip = $this->getUserIp($this->store);
+
+        $body = [
+            'Currency' => $this->groupTransaction->getCurrency(),
+            'AmountDebit' => 0,
+            'AmountCredit' => $this->groupTransaction->getRemainingAmount(),
```

Comment:
Here shouldn't be AmountCredit => $order->getGrandTotal() - $this->groupTransaction->getRemainingAmount() ?

Reply:
here we are refunding the partial payment of a giftcard/buckaroo voucher, some giftcards can be partially refunded so i take the remaining amount that has not been refunded